src/darwin/Framework/CHIP/CHIPDeviceControllerStartupParams.h

```diff
@@ -140,23 +157,68 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * * If nil:
  *
- *   * If there is an existing intermediate certificate, and it matches the
- *     nocSigner public key, the existing intermediate certificate will be used.
+ *   * If nocSigner is not nil, there is an existing intermediate certificate,
+ *     and it matches the nocSigner public key, the existing intermediate
+ *     certificate will be used.
  *
  *   * Otherwise the fabric will not use an intermediate certificate.  This
  *     allows switching from using an intermediate CA to not using one.
  *
  */
 @property (strong, nonatomic, nullable) NSData * intermediateCertificate;
 
+/**
+ * Operational certificate, in X.509 DER form, to use.
+ *
+ * If not nil, will be used as the operational certificate.  In this case
+ * operationalKeypair must not be nil.
+ *
+ * If nil, an operational certificate will be determined as described in the
+ * documentation for nodeId.
+ */
+@property (strong, nonatomic, nullable, readonly) NSData * operationalCertificate;
+
+/**
+ * Operational keypair to use.  If operationalCertificate is not nil, the public
+ * key must match operationalCertificate.
+ *
+ * If not nil, and if operationalCertificate is nil, a new operational
+ * certificate will be generated for the given operationalKeypair.  The node id
+ * will for that certificated will be determined as described in the
+ * documentation for nodeId.
+ */
+@property (strong, nonatomic, nullable) id<CHIPKeypair> operationalKeypair;
+
 - (instancetype)init NS_UNAVAILABLE;
 
 /**
+ * Prepare to initialize a controller given a keypair to use for signing
+ * operational certificates.
+ *
  * fabricId must be set to a valid (i.e. nonzero) value.
  *
  * ipk must be 16 bytes in length
```

Comment:
Watch-out, as this may not be great if you need more than one IPK or use the epoch features. The API won't allow it. Since you don't allow direct access to group data provider, perhaps allow passing a list of up to 3 <startTime, epoch key>.

Also note that this is an *epoch* IPK not the operational form

Reply:
Right, that is pre-existing.  Need to think through how to handle IPK rotation; this presumably matters for cases when the controller is the CASE responder, right?